website/docs/help-and-support/troubleshooting-guide/widget-errors.md

```diff
@@ -1,38 +1,84 @@
+This section would help you troubleshoot common widget errors on the Appsmith platform.
 
-## JSON Form Errors
+## JSON form errors
 
 You may see the below errors when working with a [JSON Form ](../../reference/widgets/json-form)widget.
 
-### **Source data exceeds 50 fields**
+### Source data exceeds 50 fields
 
 You see an error message **Source data exceeds 50 fields. Please update the source data** when you try to bind the query/API response to the [source data ](../../reference/widgets/json-form#source-data)property.
 
-**Error message:**
+#### Error message
 
 Source data exceeds 50 fields. Please update the source data.
 
 ![When the data had more than 50 fields](</img/Troubleshooting__Widget_Errors__JSON_Form_Errors__Source_Exceeds_50_Fields.png>)
 
-### Cause
+#### Cause
 
 The problem can be caused when you are trying to bind:
 
 * A large array of multiple JSON objects
 * A huge JSON object which has a lot of fields
 * The whole query data rather than a selected row or triggered row in a table
 
-### Solution
+#### Solution
 
 To determine if the problem is caused due to:
 
-* **A large array or a huge JSON object** - You can relook at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
-* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either selected row / triggered row to bind.
+* **A large array or a huge JSON object** - You can re-look at the data and evaluate the need to display all the data on UI, as it would be painful for your users to navigate more than 50 fields.
+* **The whole query response that you bound to the source data** - You recheck the source data field you are trying to bind and select either the selected row / triggered row to bind.
 
 Once you have figured out the new structure for the data, head to the [source data](../../reference/widgets/json-form#source-data) field to make changes.
 
-:::info
-If you still can't see a JSON form and need help debugging an error, please get in touch with us on our [Discord Server](https://discord.com/invite/rBTTVJp) or email us at support@appsmith.com.
-:::
\ No newline at end of file
+## Query/API/JS object execution errors
+You may see below errors when executing an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS object](../../core-concepts/writing-code/javascript-editor-beta/) using a widget property.
+
+### Sync fields can't execute async framework actions
+
+You may see this error when binding an [API](../../core-concepts/connecting-to-data-sources/authentication/)/[Query](../../core-concepts/data-access-and-binding/querying-a-database/)/[JS Object](../../core-concepts/writing-code/javascript-editor-beta/) execution to a widget property that expects data, and can't be used to trigger an action. 
```

Comment:
Just confirming that this error is seen in Query and API also? or just JSObjects @bharath31

Reply:
For all the three. If the Query/API is executed the name of the same will be replaced in {{action}}. However, if it's JS object function then it is Promise() that is shown. CC: @Rishabh-Rathod - Please confirm.

Reply:
The reason I brought this up is because usually users are aware how to bind the result of query from datasource or API. they know that using .data gives them the result

As for JS Object, when the function is sync, they can directly use {{JSObject.functionName}} in the widget and it works fine. On support queries, I have only ever seen users stuck on this error when the function is async, in which case they have to invoke the function with a .data property.